src/lib/sql/sql_identifier_resolver.cpp

```diff
@@ -48,19 +53,25 @@ std::shared_ptr<AbstractExpression> SQLIdentifierResolver::resolve_identifier_re
   return matching_expressions[0];
 }
 
-const std::optional<SQLIdentifier> SQLIdentifierResolver::get_expression_identifier(
+const std::vector<SQLIdentifier>& SQLIdentifierResolver::get_expression_identifiers(
     const std::shared_ptr<AbstractExpression>& expression) const {
   auto entry_iter = std::find_if(_entries.begin(), _entries.end(),
                                  [&](const auto& entry) { return *entry.expression == *expression; });
-  if (entry_iter == _entries.end()) return std::nullopt;
-  return entry_iter->identifier;
+  const auto empty_vector = std::make_shared<std::vector<SQLIdentifier>>();
+  if (entry_iter == _entries.end()) return *empty_vector;
```

Comment:
This returns a reference to a local variable, which is subsequently destroyed. Did none of our sanitizers catch this?

Reply:
I think I'd be fine with return-by-value.

Reply:
Neither `lint.sh` not `format.sh` did something about this. Shouldn't the object be kept until there is no more reference to it because it comes from a shared pointer?

Reply:
`*empty_vector` does not increase the counter, so there is exactly one `shared_ptr` pointing to `empty_vector`. When that `shared_ptr` goes out of scope, `*empty_vector` is broken.

My approach for things like this is to have a `static` empty object and return a const reference to that one.

Reply:
> Neither lint.sh not format.sh did something about this

I was referring to valgrind, ASAN and UBSAN, but since this PR doesn't have FullCI enabled yet, those haven't actually run yet.

@mrks' solution is fine (though I always wondered whether there might be some performance hit when using it in multi-threaded scenarios)

Reply:
You don't get any software level parallelization guarantees. If you want to access it from multiple threads, you will have to synchronize the access yourself. As the object is not modified, there should also be no cache line synchronization necessary. A `const` wouldn't hurt though.

Reply:
Oh, but you do, right? 

https://stackoverflow.com/a/17713799

"...static storage duration for a function local means that only one instance of that local exists in the program" 

"The thread safety bits make it get a bit more complicated..."

But looking at the assembly in the SO answer, the primary guard variable is not synchronized as far as I can tell, so the overhead *should* be minimal

Reply:
I chose to do it return-by-value because the `SQLIdentifier`s are not part of the resolver anymore, but they are created ad hoc.